rubullet/src/client.rs

```diff
@@ -0,0 +1,3254 @@
+                ptr::null_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+                &actual_state_q as _,
+                ptr::null_mut(),
+                ptr::null_mut(),
+            );
+
+            assert!(!actual_state_q.is_null());
+
+            let tx = *actual_state_q;
+            let ty = *(actual_state_q.offset(1));
+            let tz = *(actual_state_q.offset(2));
+
+            let rx = *(actual_state_q.offset(3));
+            let ry = *(actual_state_q.offset(4));
+            let rz = *(actual_state_q.offset(5));
+            let rw = *(actual_state_q.offset(6));
+
+            let tra = Translation3::new(tx, ty, tz);
+            let rot = Quaternion::new(rw, rx, ry, rz);
+
+            Ok(Isometry3::from_parts(
+                tra,
+                UnitQuaternion::from_quaternion(rot),
+            ))
+        }
+    }
+    /// You can reset the position and orientation of the base (root) of each object.
+    /// It is best only to do this at the start, and not during a running simulation,
+    /// since the command will override the effect of all physics simulation.
+    /// The linear and angular velocity is set to zero.
+    /// You can use [reset_base_velocity](`Self::reset_base_velocity()`)
+    /// to reset to a non-zero linear and/or angular velocity.
+    /// # Arguments
+    /// * `body` - the [`BodyId`](`crate::types::BodyId`), as returned by [`load_urdf`](`Self::load_urdf()`) etc.
+    /// * `pose` - reset the base of the object at the specified position in world space coordinates
+    pub fn reset_base_transform(&mut self, body: BodyId, pose: &Isometry3<f64>) {
+        unsafe {
+            let command_handle = ffi::b3CreatePoseCommandInit(self.handle.as_ptr(), body.0);
+            ffi::b3CreatePoseCommandSetBasePosition(
+                command_handle,
+                pose.translation.x,
+                pose.translation.y,
+                pose.translation.z,
+            );
+            ffi::b3CreatePoseCommandSetBaseOrientation(
+                command_handle,
+                pose.rotation.i,
+                pose.rotation.j,
+                pose.rotation.k,
+                pose.rotation.w,
+            );
+            let _status_handle =
+                ffi::b3SubmitClientCommandAndWaitStatus(self.handle.as_ptr(), command_handle);
+        }
+    }
+    /// You get access to the linear and angular velocity of the base of a body.
+    /// # Arguments
+    /// * `body` - the [`BodyId`](`crate::types::BodyId`), as returned by [`load_urdf`](`Self::load_urdf()`) etc.
+    /// # Return
+    ///  Returns a result which is either a velocity array of size 6 (linear velocity, angular velocity)
+    ///  or an Error
+    /// # See also
+    /// [reset_base_velocity](`Self::reset_base_velocity()`) to reset a base velocity and for examples
+    pub fn get_base_velocity(&mut self, body: BodyId) -> Result<[f64; 6], Error> {
```

Comment:
done. But I did it with a vector6 and with methods to get the linear and angular part, because right now it is easy to split vectors up, but hard to combine them.

```rust
pub struct Velocity(Vector6<f64>);

impl Velocity {
    /// returns the linear velocity (x,y,z)
    pub fn get_linear_velocity(&self) -> Vector3<f64> {
        self.0.fixed_rows::<U3>(0).into()
    }
    /// returns the angular velocity (wx,wy,wz)
    pub fn get_angular_velocity(&self) -> Vector3<f64> {
        self.0.fixed_rows::<U3>(3).into()
    }
    /// converts the velocity to a Vector6 (x,y,z,wx,wy,wz)
    pub fn to_vector(&self) -> Vector6<f64> {
        self.0
    }
}
```
I did the same for the jacobian